### PR TITLE
All Reviews block

### LINKED ISSUE #847: New Block: All reviews
Separated from https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/589

A simple block to list all reviews, rather than those from a category or single product.

### PR DESCRIPTION
**Pull request by Aljullu**
Fixes #847.

This PR:
* Creates an _All Reviews_ block.
* Makes the three review blocks share the same frontend script.
* Moves some editor logic from each block to shared files.

### Screenshots

image

### How to test the changes in this Pull Request:

1. Create a post and add an _All Reviews_ block.
2. Try modifying attributes and verify they are honored in the editor and the frontend.
3. Since this PR moves some code from other reviews blocks to shared files, it would be good to test _Reviews by Product_ and _Reviews by Category_ blocks to.

### Changelog

> A new block named 'All Reviews' was added in order to display a list of reviews from all products and categories of your store.

### CODE REVIEW

File: assets/js/blocks/reviews/all-reviews/edit.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		);
+	};
+
+	return (
+		<Fragment>
+			{ getInspectorControls() }
+			<EditorContainerBlock
+				attributes={ attributes }
+				className="wc-block-all-reviews"
+				icon={ <Gridicon icon="grid" /> }
```
Comment (mikejolley) [inline assets/js/blocks/reviews/all-reviews/edit.js]: Pinged on Slack - suggest using `list`

File: assets/js/blocks/reviews/all-reviews/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import sharedAttributes from '../attributes';
+import save from '../save.js';
+
+/**
+ * Register and run the "All Reviews" block.
+ */
+registerBlockType( 'woocommerce/all-reviews', {
+	title: __( 'All Reviews', 'woo-gutenberg-products-block' ),
+	icon: {
+		src: <Gridicon icon="grid" />,
```
Comment (mikejolley) [inline assets/js/blocks/reviews/all-reviews/index.js]: Same as above, suggest a different icon.

File: assets/js/blocks/reviews/all-reviews/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		foreground: '#96588a',
+	},
+	category: 'woocommerce',
+	keywords: [ __( 'WooCommerce', 'woo-gutenberg-products-block' ) ],
+	description: __(
+		'Show product reviews from all products and categories.',
```
Comment (mikejolley) [inline assets/js/blocks/reviews/all-reviews/index.js]: `Shows a list of all product reviews.`?

Reviews are only for products, not categories. And the name already says they are for products :)

File: assets/js/blocks/reviews/all-reviews/no-reviews-placeholder.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ */
+import { __ } from '@wordpress/i18n';
+import { Placeholder } from '@wordpress/components';
+import Gridicon from 'gridicons';
+
+const NoCategoryReviewsPlaceholder = () => {
+	return (
+		<Placeholder
+			className="wc-block-all-reviews"
+			icon={ <Gridicon icon="grid" /> }
```
Comment (mikejolley) [inline assets/js/blocks/reviews/all-reviews/no-reviews-placeholder.js]: icon

File: assets/js/blocks/reviews/all-reviews/no-reviews-placeholder.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+			label={ __( 'All Reviews', 'woo-gutenberg-products-block' ) }
+		>
+			{ __( 'This block lists reviews from all products and categories. Currently your store does not have any reviews yet, but they will show up here when it does.', 'woo-gutenberg-products-block' ) }
```
Comment (mikejolley) [inline assets/js/blocks/reviews/all-reviews/no-reviews-placeholder.js]: `Currently your store does not have any reviews yet, but they will show up here when it does.`

I would rephrase this as:

`Your store does not have any reviews yet, but they will show up here when it does.`

File: assets/js/blocks/reviews/edit-utils.js [hunk 6]
```diff
@@ -14,6 +16,21 @@ import { ENABLE_REVIEW_RATING, SHOW_AVATARS } from '@woocommerce/settings';
  */
 import ToggleButtonControl from '../../components/toggle-button-control';
 
+export const getBlockControls = ( editMode, setAttributes ) => (
+	<BlockControls>
+		<Toolbar
+			controls={ [
+				{
+					icon: 'edit',
+					title: __( 'Edit' ),
```
Comment (mikejolley) [inline assets/js/blocks/reviews/edit-utils.js]: Does this need a text domain?

File: assets/js/blocks/reviews/editor-block.js [hunk 7]
```diff
@@ -69,9 +56,7 @@ EditorBlock.propTypes = {
 	 * The attributes for this block.
 	 */
 	attributes: PropTypes.object.isRequired,
-	/**
-	 * From withComponentId.
-	 */
+	// from withComponentId
```
Comment (mikejolley) [inline assets/js/blocks/reviews/editor-block.js]: Just a question, why do we use a different comment format just for this item?

File: assets/js/blocks/reviews/editor-container-block.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+					order={ order }
+					productId={ productId }
+					reviewsToDisplay={ reviewsOnPageLoad }
+				/>
+			</div>
+		);
+	}
+}
+
+EditorContainerBlock.propTypes = {
```
Comment (mikejolley) [inline assets/js/blocks/reviews/editor-container-block.js]: Could use static propTypes for this class if wanted, now thats available.

File: assets/js/blocks/reviews/editor-block.js (hunk 7 above)
Comment (Aljullu) [reply assets/js/blocks/reviews/editor-block.js]: We are using this format for most props added by HOCs, so I tried to keep it consistent:
Screenshot

I don't know what was the rationale when that decision was taken. An idea is so that this kind of comments are skipped if some day we use JSDoc or similar tools (JSDoc takes `/** */` but ignores `//`).
